bee-message/src/output/alias.rs

```diff
@@ -170,7 +192,18 @@ impl AliasOutputBuilder {
             unlock_conditions,
             feature_blocks,
             immutable_feature_blocks,
-        })
+        };
+
+        output.amount = match (self.amount, self.byte_cost_config) {
+            (Some(amount), None) => amount,
+            (None, Some(byte_cost_config)) => Output::Alias(output.clone())
+                .byte_cost(&byte_cost_config)
+                .try_into()
+                .map_err(Error::InvalidOutputAmount)?,
+            _ => unreachable!(),
```

Comment:
This makes me thing that `amount` and `byte_cost_config` should be something like `Result<OutputAmount, ByteCostConfig>`

Reply:
I proposed something very similar to @Thoralf-M but it looked like it was better in terms of API to have 2 different methods.

Reply:
I mean, we could actually do a mix of both. Have the two methods AND the enum.

Reply:
Yeah I'd keep the methods but use an enum internally so these illegal states cannot be represented at all.